source/Outfit.cpp

```diff
@@ -384,6 +399,86 @@ const string &Outfit::Description() const
 
 
 
+uint32_t Outfit::PassengerBunks() const
+{
+	uint32_t passengers = bunks;
+	for(auto bunkType : bunkTypes)
+	{
+		if(GameData::GetBunkType(bunkType.first).CanHoldPassengers())
+			passengers += bunkType.second;
+	}
+	return passengers;
+}
+
+
+
+uint32_t Outfit::CrewBunks() const
+{
+	uint32_t crew = bunks;
+	for(auto bunkType : bunkTypes)
+		if(GameData::GetBunkType(bunkType.first).CanHoldCrew())
+			crew += bunkType.second;
+	return crew;
+}
+
+
+
+uint32_t Outfit::FreePassengerBunks(uint32_t shipCrew) const
+{
+	uint32_t passengerSpace = 0;
+	uint32_t dualUseSpace = bunks;
+	uint32_t crewSpace = 0;
+	for(auto bunkType : bunkTypes)
+	{
+		if(GameData::GetBunkType(bunkType.first).CanHoldPassengers()
+			&& !GameData::GetBunkType(bunkType.first).CanHoldCrew())
+			passengerSpace += bunkType.second;
+		else if(GameData::GetBunkType(bunkType.first).CanHoldPassengers()
+			&& GameData::GetBunkType(bunkType.first).CanHoldCrew())
+			dualUseSpace += bunkType.second;
+		else if(!GameData::GetBunkType(bunkType.first).CanHoldPassengers()
+			&& GameData::GetBunkType(bunkType.first).CanHoldCrew())
+			crewSpace += bunkType.second;
+	}
+	if(shipCrew <= crewSpace)
+		return passengerSpace + dualUseSpace;
+	else if(shipCrew <= crewSpace + dualUseSpace)
+		return passengerSpace + crewSpace + dualUseSpace - shipCrew;
+	return 0;
+}
+
+
+
+uint32_t Outfit::FreeCrewBunks(uint32_t shipCrew) const
+{
+	uint32_t dualUseSpace = bunks;
+	uint32_t crewSpace = 0;
+	for(auto bunkType : bunkTypes)
+	{
+		if(GameData::GetBunkType(bunkType.first).CanHoldPassengers()
+			&& GameData::GetBunkType(bunkType.first).CanHoldCrew())
+			dualUseSpace += bunkType.second;
+		else if(!GameData::GetBunkType(bunkType.first).CanHoldPassengers()
+			&& GameData::GetBunkType(bunkType.first).CanHoldCrew())
+			crewSpace += bunkType.second;
+	}
+	if(shipCrew <= crewSpace + dualUseSpace)
+		return crewSpace + dualUseSpace - shipCrew;
+	return 0;
+}
+
+
+
+uint32_t Outfit::GetBunkType(string typeName) const
+{
+	auto finder = bunkTypes.find(typeName);
+	if(finder != bunkTypes.end())
+		return finder->second;
+	return 0;
+}
```

Comment:
What I had in mind i initially was to put all of this into the new class but I realize that could need another class entirely
Because of that I'd say its not something to do right now

However, should this not be part of cargohold?